core/controller/src/main/scala/org/apache/openwhisk/core/controller/Activations.scala

```diff
@@ -219,8 +223,15 @@ trait WhiskActivationsApi extends Directives with AuthenticatedRouteProvider wit
    * - 500 Internal Server Error
    */
   private def fetchLogs(context: UserContext, docid: DocId)(implicit transid: TransactionId) = {
-    getEntityAndProject(
+    getEntityAndProjectLog(
       activationStore.get(ActivationId(docid.asString), context),
-      (activation: WhiskActivation) => logStore.fetchLogs(activation, context).map(_.toJsonObject))
+      docid,
+      disableStoreResultConfig,
+      (namespace: String,
+       activationId: String,
```

Comment:
about my earlier comment regarding the type `ActivationId`, notice on line 223/227 an instance of `ActivationId` is created. I'd rather that's used everywhere for prudence.